### ISSUE TITLE
plugin in Atlassian Marketplace?

### ISSUE THREAD

**Issue by gliptak** [palantir/stash-codesearch-plugin#9]
Is this plugin uploaded to the Atlassian Marketplace? Thanks

https://marketplace.atlassian.com/plugins/app/stash/

(Reading the pom.xml, the plugin supports 3.6+ of Stash only?)

**Comment by terabyte** [comment]
We tried to upload it to the marketplace, but they pushed back a bit because the default settings were a little strange (i.e. by default it is turned off and you need to turn it on) and a few other minor changes were needed.  I am pretty sure this just never happened due to lack of time.  The plugin definitely supports the revision in the POM, but may or may not work with older or newer versions.  The best way to know is to clone it, change the version in the pom, then see if it compiles and the tests pass.

**Comment by gliptak** [comment]
@terabyte Would you try submitting again and post the asks as Issues here (so they can be worked, except of course the default "on" ...)? Thanks

**Comment by BenjaminDSmithy** [comment]
This may be a month+ late but I thought I'd mention it here.

So it appears that Mohamicorp has released a **paid** Stash Add-on very similar if not identical to this one called [Search for Stash](https://marketplace.atlassian.com/plugins/com.mohamicorp.stash.search-for-stash), which might breach an open-source license on this repository if it is identical...

You can find a [comment](https://jira.atlassian.com/browse/STASH-2851?focusedCommentId=745150&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-745150) related to this topic on the JIRA issue [STASH-2851](https://jira.atlassian.com/browse/STASH-2851)

**Disclaimer:** I have not personally cross-compared these 2 Stash Add-ons, so I'm not in the position to point fingers by any means but I thought I'd at least bring it to your (@terabyte) attention nonetheless.

**Comment by sl33kr** [comment]
I believe they are based on the same code base.

The index that "Search for Stash" creates is stash-codesearch and the URL for the settings page is plugins/servlet/codesearch...

However I believe under the Apache 2.0 License this is allowed to happen? However they do seem to be in breach of the license in that they don't
- Include original Copyright
- Include a license
- State if any changes have been made

![stashcodesearchglobal settingsstash](https://cloud.githubusercontent.com/assets/1491080/8950049/7b6c6956-35ff-11e5-8152-cf105c7bcbb4.png)
![searchforstashglobalsettingsstash](https://cloud.githubusercontent.com/assets/1491080/8950050/7b6c7e5a-35ff-11e5-82bc-ec3757e562b4.png)

**Comment by terabyte** [comment]
I will bring this to the attention of Palantir's Legal Department.  Thanks for pointing this out guys!

**Comment by gliptak** [comment]
@terabyte  Also would you inquire with the current team if they are interested to publish from this repo (or should we publish from a forked repo to Atlassian)? Thanks

**Comment by terabyte** [comment]
Yeah, so as I've said elsewhere (but not yet said here) I no longer work for Palantir and the team I was on pretty much doesn't exist anymore, so I'm having trouble getting much traction at Palantir.  I'm basically a weekend of boredom away from forking every plugin I wrote there and taking it on myself.  I would greatly prefer Palantir maintain these plugins, but if they won't, I will.

**Comment by terabyte** [comment]
@atrocities ^^

**Comment by gliptak** [comment]
@terabyte Please point us to the new repos when available :)

**Comment by gliptak** [comment]
@terabyte Did you proceed with uploading this plugin? Could we have 3.x and 4,x supporting versions?

**Comment by terabyte** [comment]
@gliptak I haven't uploaded a binary release yet (and I won't be able to, on the plugin marketplace, because palantir owns the marketplace entry), but you should be able to build the release on any linux machine with a 1.8 jdk on your path and JAVA_HOME set by running `./bin/invoke-sdk.sh clean test package`.

https://github.com/terabyte/stash-codesearch-plugin/

I've added travis-ci builds of both the stash-3.x-backports branch and the master branch (which is stash 4.0 compatible) and I accept PRs to either branch if you need any changes.  Many thanks!

**Comment by terabyte** [comment]
I may tag a 2.0 release and upload the jars to github, to make things easier for folks, but I haven't decided for certain yet.
